apps/datahub/src/app/organization/organization-details/organization-details.component.ts

```diff
@@ -0,0 +1,189 @@
+import { UiSearchModule } from '@geonetwork-ui/ui/search'
+import { SearchFacade } from '@geonetwork-ui/feature/search'
+import { BehaviorSubject, Observable, of, Subscription, switchMap } from 'rxjs'
+import { UiDatavizModule } from '@geonetwork-ui/ui/dataviz'
+import { RouterLink } from '@angular/router'
+import { ROUTER_ROUTE_SEARCH } from '@geonetwork-ui/feature/router'
+import { OrganizationsServiceInterface } from '@geonetwork-ui/common/domain/organizations.service.interface'
+import { UiWidgetsModule } from '@geonetwork-ui/ui/widgets'
+
+@Component({
+  selector: 'datahub-organization-details',
+  templateUrl: './organization-details.component.html',
+  styleUrls: ['./organization-details.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    NgIf,
+    ButtonComponent,
+    MatIconModule,
+    TranslateModule,
+    CarouselComponent,
+    BlockListComponent,
+    LetDirective,
+    LinkCardComponent,
+    NgForOf,
+    PreviousNextButtonsComponent,
+    UiElementsModule,
+    UiSearchModule,
+    MaxLinesComponent,
+    UiDatavizModule,
+    RouterLink,
+    UiWidgetsModule,
+    NgClass,
+  ],
+})
+export class OrganizationDetailsComponent
+  implements OnInit, AfterViewInit, OnDestroy, OnChanges
+{
+  protected readonly Error = Error
+  protected readonly ErrorType = ErrorType
+  protected readonly ROUTER_ROUTE_SEARCH = ROUTER_ROUTE_SEARCH
+
+  protected get pages() {
+    return new Array(this.totalPages).fill(0).map((_, i) => i + 1)
+  }
+
+  lastPublishedDatasets$: Observable<CatalogRecord[]> = of([])
+
+  subscriptions$: Subscription = new Subscription()
+
+  isOrganizationsLoading = true
+
+  totalPages = 0
+  currentPage = 1
+  isFirstPage = this.currentPage === 1
+  isLastPage = false
+
+  organizationHasChanged$ = new BehaviorSubject<void>(undefined)
+
+  @Input() organization: Organization
+  @Input() paginationContainerClass = 'w-full bottom-0 top-auto'
+
+  @ViewChild(BlockListComponent) list: BlockListComponent
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private searchFacade: SearchFacade,
+    private organizationsService: OrganizationsServiceInterface
+  ) {}
+
+  ngOnInit(): void {
+    this.searchFacade.setPageSize(3)
+
+    this.lastPublishedDatasets$ = this.organizationHasChanged$.pipe(
+      switchMap(() => {
+        return this.organizationsService
+          .getFiltersForOrgs([this.organization])
+          .pipe(
+            switchMap((filters) => {
+              return this.searchFacade
+                .setFilters(filters)
+                .setSortBy(['desc', 'changeDate']).results$
+            })
+          )
+      })
+    )
+
+    this.manageSubscriptions()
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['organization']) {
+      this.organizationHasChanged$.next()
+    }
+  }
+
+  ngAfterViewInit() {
+    // this is required to show the pagination correctly
```

Comment:
So this one is a comment from Olivia. I think it is usefull because it tells to the dev that may wondering why we are doing the `this.changeDetector.detectChanges() `... So... Should I remove it? What about the one in `apps/datahub/src/app/record/record-otherlinks/record-otherlinks.component.ts`

Reply:
Yes it's useful because as you say, it's not straightforward to understand what this is for. On the opposite, your comment on the e2e was describing the code (and accessory, a workflow that is easy to understand), but not giving us reasons about why you wrote that. For instance, if you had written "Go to the organization page first, because we need to display the organizations list to access the page", then it would have been useful.

Reply:
(But again I think the workflow of going to the org page to click on an org item is easy to understand :slightly_smiling_face: )

Reply:
But yes this one can stay, on second thought